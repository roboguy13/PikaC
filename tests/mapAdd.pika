%generate mapAdd

Sll : layout[-x](List);
Sll Nil := emp;
Sll (Cons head tail) :=
  exists +nxt.
  x :-> head **
  (x+1) :-> nxt **
  Sll tail [nxt];

mapAdd : Int -> Sll -> Sll;
mapAdd n Nil := Nil [Sll];
mapAdd n (Cons a as) := Cons (a + n) (mapAdd n as) [Sll];

%test "mapAdd1" Sll:
  mapAdd 1 (Cons 10 (Cons 20 (Cons 30 (Nil [Sll]) [Sll]) [Sll]) [Sll]);
