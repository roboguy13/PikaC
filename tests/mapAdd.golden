*** Running test mapAdd1
(11 (21 (31 ())))

